iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -485,20 +489,17 @@ inline
 {
     uint64_t tSize = internal::GetSize<T>::call(t);
     const char* tData = internal::GetData<T>::call(t);
-    if (Capacity < (m_rawstringSize + tSize))
+    uint64_t clampedTSize = std::min(Capacity - m_rawstringSize, tSize);
+
+    std::memcpy(&(m_rawstring[m_rawstringSize]), tData, clampedTSize);
+    if (tSize > clampedTSize)
     {
         std::cerr << "The last " << tSize - Capacity + m_rawstringSize << " characters of " << tData
                   << " are truncated, because the length is larger than the capacity." << std::endl;
```

Comment:
this could theoretically be optimized by replacing `tSize - Capacity + m_rawstringSize` with `tSize - clampedTSize`